## Q1
Is gender independent of education level? A random sample of 395 people were
surveyed and each person was asked to report the highest education level they
obtained. The data that resulted from the survey is summarized in the following table:
High School Bachelors Masters Ph.d. Total
Female 60 54 46 41 201
Male 40 44 53 57 194
Total 100 98 99 98 395
Question: Are gender and education level dependent at 5% level of significance? In
other words, given the data collected above, is there a relationship between the
gender of an individual and the level of education that they have obtained?

In [1]:
from scipy.stats import chi2_contingency
import pandas as pd
import numpy as np

In [4]:
data=pd.DataFrame([ [60 ,54, 46, 41, 201], [40, 44, 53, 57,194],[100, 98, 99, 98, 395]] , index=['Female','Male','Total'] , columns=['High School', 'Bachelors', 'Masters', 'Ph.d.', 'Total'])  
 

In [5]:
data

High School  Bachelors  Masters  Ph.d.  Total
Female           60         54       46     41    201
Male             40         44       53     57    194
Total           100         98       99     98    395

In [11]:
d1=data.drop(columns='Total', index='Total')

In [12]:
d1

High School  Bachelors  Masters  Ph.d.
Female           60         54       46     41
Male             40         44       53     57

In [13]:
chi2_contingency(d1)

(8.006066246262538,
 0.045886500891747214,
 3,
 array([[50.88607595, 49.86835443, 50.37721519, 49.86835443],
        [49.11392405, 48.13164557, 48.62278481, 48.13164557]]))

In [15]:
dof=chi2_contingency(d1)[2]
dof

3

Acc to the question to see if their is a relationship between gender and level of education we perform chi square test.
NULL HYPOTHESIS-They are independent of each other
Alt Hypothesis- They are dependent on each other.

We will compare the value of the test statistic to the critical value (X2 at aplha) of  with 
degree of freedom = (r - 1) (c - 1), 
and reject the null hypothesis if X2> X2 at aplha. ie calculated> theoritical

In [20]:
chivalue=chi2_contingency(d1)[0]
chivalue

8.006066246262538

In [16]:
from scipy.stats import chi2
significance = 0.05
p = 1 - significance
dof = chi2_contingency(d1)[2]
critical_value = chi2.ppf(p, dof)
critical_value

7.814727903251179

In [17]:
p = chi2.cdf(critical_value, dof)
p

0.9500000000000001

The critical value of  with 3 degree of freedom is 7.81 . Since chi value > critical value, we reject the null hypothesis and conclude that the education level depends on gender at a 5% level of significance

In [21]:
chi, pval, dof, exp = chi2_contingency(d1)
significance = 0.05
print('p-value=%.6f, significance=%.2f\n' % (pval, significance))

p-value=0.045887, significance=0.05



 Since pval < significance:
At o.05 level of significance, we reject the null hypotheses and accept H1.

## Using the following data, perform a oneway analysis of variance using α=.05. Write up the results in APA format.

[Group1: 51, 45, 33, 45, 67]
[Group2: 23, 43, 23, 43, 45]
[Group3: 56, 76, 74, 87, 56] 

In [22]:
import scipy.stats as stats

In [23]:
Group1=[ 51, 45, 33, 45, 67]
Group2=[ 23, 43, 23, 43, 45]
Group3=[ 56, 76, 74, 87, 56]

In [26]:
F, p = stats.f_oneway(Group1, Group2, Group3)
print("F value is={} , pvalue={}".format(F,p))

F value is=9.747205503009463 , pvalue=0.0030597541434430556


The null hypothesis (H0) of ANOVA is that there is no difference among group means. The alternate hypothesis (Ha) is that at least one group differs significantly from the overall mean of the dependent variable

In [30]:
if p<0.05:
    print("null hypothesis is rejected")
else:
    print("accepted")

null hypothesis is rejected


## Calculate F Test for given 10, 20, 30, 40, 50 and 5,10,15, 20, 25. For 10, 20, 30, 40, 50:

In [32]:
g1=[10, 20, 30, 40, 50]
g2=[5, 10,15,20,25]

m1=np.mean(g1)
m2=np.mean(g2)

v1=np.var(g1)
v2=np.var(g2)

In [36]:
print("Fvalue is:",v1/v2)

Fvalue is: 4.0
